xc/device.ml

```diff
@@ -1392,6 +1392,82 @@ module Vkbd = struct
 
 end
 
+module Vusb = struct
+  let vusb_controller_plug ~xs ~domid ~driver ~driver_id =
+    let is_running = Qemu.is_running ~xs domid in
+    match is_running with
+    | true  -> (
+      let open Qmp in
+      let qmp_cmd = Command (None, Device_add (driver, driver_id, None)) in
+      qmp_write domid qmp_cmd
+    )
+    | _ ->()
+
+  let vusb_plug ~xs ~domid ~id ~hostbus ~hostport ~version =
+    debug "Vusb plugged: vusb device plugged ";
+    let get_bus v =
+      if String.startswith "1" v then
+         "usb-bus.0"
+      else begin
+          (* Here plug usb controller according to the usb version*)
+          let usb_controller_driver = "usb-ehci" in
+          let driver_id = "ehci" in
+          vusb_controller_plug ~xs ~domid ~driver:usb_controller_driver ~driver_id;
+          "ehci.0"
+      end
+    in
+    let is_running = Qemu.is_running ~xs domid in
+    match is_running with
+    | true  -> (
+      let open Qmp in
+      let driver = "usb-host" in
+      let qmp_cmd = Command (None, Device_add (driver, id, Some (get_bus version, hostbus, hostport))) in
+      qmp_write domid qmp_cmd
+    )
+    | _ ->()
+
+  let vusb_unplug ~xs ~domid ~id =
+    debug "Vusb unplugged: vusb device unplugged";
+    let is_running = Qemu.is_running ~xs domid in
+    match is_running with
+    | true  -> (
+      let open Qmp in
+      let qmp_cmd = Command (None, Device_del (id)) in
+      debug "qmp unplug usb device";
```

Comment:
Include more information about the device - otherwise the log line is useless

Reply:
yes,  here the debug is useless

Reply:
updated. Thanks